File: 7Modules/Jacob/frontend/src/types/order.ts
export type OrderItemCreate = {
  category_id: number;
  part_number: string;
  part_description?: string;
  quantity: number;
};

export type OrderCreatePayload = {
  items: OrderItemCreate[];
  urgency: "standard" | "urgent" | "critical";
  required_delivery_date?: string;
};

export type OrderAssignment = {
  id: number;
  order_item_id: number;
  supplier_id: number;
  catalog_id: number;
  assigned_price?: number | null;
  match_score?: number | null;
  status: "PROPOSED" | "ACCEPTED" | "REJECTED" | "FULFILLED";
  created_at?: string | null;
  supplier_business_name?: string | null;
  supplier_user_id?: number | null;
  supplier_latitude?: number | null;
  supplier_longitude?: number | null;
  distance_km?: number | null;
};

export type OrderItem = {
  id: number;
  order_id: number;
  category_id: number;
  part_number: string;
  part_description?: string | null;
  quantity: number;
  status:
    | "PENDING"
    | "MATCHED"
    | "CONFIRMED"
    | "DISPATCHED"
    | "IN_TRANSIT"
    | "DELIVERED"
    | "CANCELLED";
  category_name?: string | null;
  assignments: OrderAssignment[];
};

export type OrderHistoryEntry = {
  order_item_id?: number | null;
  from_status?: string | null;
  to_status: string;
  timestamp?: string | null;
  changed_by?: number | null;
};

export type OrderRecord = {
  id: number;
  buyer_id: number;
  status:
    | "PLACED"
    | "MATCHED"
    | "CONFIRMED"
    | "DISPATCHED"
    | "IN_TRANSIT"
    | "DELIVERED"
    | "CANCELLED";
  urgency: "standard" | "urgent" | "critical";
  required_delivery_date?: string | null;
  created_at?: string | null;
  updated_at?: string | null;
  buyer_factory_name?: string | null;
  buyer_user_id?: number | null;
  buyer_latitude?: number | null;
  buyer_longitude?: number | null;
  total_items: number;
  total_value: number;
  items: OrderItem[];
  history: OrderHistoryEntry[];
};

export type OrdersListResponse = {
  items: OrderRecord[];
  page: number;
  page_size: number;
  total: number;
};

File: Original/backend/schemas/order.py
from datetime import datetime
from typing import List, Optional

from pydantic import Field

from backend.schemas.inventory import ORMBaseModel


class OrderItemCreate(ORMBaseModel):
    category_id: int
    part_number: str
    part_description: Optional[str] = None
    quantity: int


class OrderCreate(ORMBaseModel):
    items: List[OrderItemCreate]
    urgency: str = "standard"
    required_delivery_date: Optional[datetime] = None


class OrderAssignmentCreate(ORMBaseModel):
    supplier_id: int
    catalog_id: int
    price: float
    score: float


class StatusTransitionRequest(ORMBaseModel):
    new_status: str


class OrderAssignmentResponse(ORMBaseModel):
    id: int
    order_item_id: int
    supplier_id: int
    catalog_id: int
    assigned_price: Optional[float] = None
    match_score: Optional[float] = None
    status: str
    created_at: Optional[datetime] = None
    supplier_business_name: Optional[str] = None
    supplier_user_id: Optional[int] = None
    supplier_latitude: Optional[float] = None
    supplier_longitude: Optional[float] = None
    distance_km: Optional[float] = None


class OrderItemResponse(ORMBaseModel):
    id: int
    order_id: int
    category_id: int
    part_number: str
    part_description: Optional[str] = None
    quantity: int
    status: str
    category_name: Optional[str] = None
    assignments: List[OrderAssignmentResponse] = Field(default_factory=list)


class OrderHistoryEntry(ORMBaseModel):
    order_item_id: Optional[int] = None
    from_status: Optional[str] = None
    to_status: str
    timestamp: Optional[datetime] = None
    changed_by: Optional[int] = None


class OrderResponse(ORMBaseModel):
    id: int
    buyer_id: int
    status: str
    urgency: str
    required_delivery_date: Optional[datetime] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    buyer_factory_name: Optional[str] = None
    buyer_user_id: Optional[int] = None
    buyer_latitude: Optional[float] = None
    buyer_longitude: Optional[float] = None
    total_items: int = 0
    total_value: float = 0.0
    items: List[OrderItemResponse] = Field(default_factory=list)
    history: List[OrderHistoryEntry] = Field(default_factory=list)


class OrdersListResponse(ORMBaseModel):
    items: List[OrderResponse]
    page: int
    page_size: int
    total: int
